File: inst/expt/test_conv.py
from cvxpy import *
import canonInterface
#from cvxpy.atoms.affine import conv

n = 3
x = Variable(n)
f = [1, 2, 3]
g = [0, 1, 0.5]
f_conv_g = [0., 1., 2.5,  4., 1.5]
expr = conv(f, x)

# Matrix stuffing.
t = Variable()
prob = Problem(Minimize(norm(expr, 1)),
               [x == g])
result = prob.solve()
# b conv.value
w = expr.value

##self.assertAlmostEqual(result, sum(f_conv_g))
##self.assertItemsAlmostEqual(expr.value, f_conv_g)

File: inst/expt/test_conv.R
library(cvxr)

n <- 3
x <- Variable(n)
f <- c(1, 2, 3)
g <- c(0, 1, 0.5)
f_conv_g <- c(0, 1, 2.5, 4, 1.5)
expr <- Conv(f, x)

## Matrix stuffing
prob <- Problem(Minimize(Norm(expr, 1)), list(x == g))
result <- solve(prob)

result$getValue(x)

result$getValue(expr)
